ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package io.ripc.protocol.tcp;
+
+public interface TcpInterceptor<I, O, II, OO> {
```

Comment:
I agree that 4 generic types are verbose but I don't see how moving two to the method level will make it any different. The method definition then becomes horrific.

Finagle makes it simpler for interceptors that do not change the types, by having a `SimpleFilter` which has the same in & out types.